stdlib/src/python/_cpython.mojo

```diff
@@ -79,7 +79,16 @@ alias newfunc = fn (PyObjectPtr, PyObjectPtr, PyObjectPtr) -> PyObjectPtr
 @value
 @register_passable("trivial")
 struct PyGILState_STATE:
+    """Represents the state of the Python Global Interpreter Lock (GIL).
+
+    This struct is used to store and manage the state of the GIL,
```

Comment:
👍 currently our use of CPython APIs aren't GIL-aware (basically, Mojo holds the GIL exclusively which isn't good).  If you'd like to start this work, ping me.  I've fixed some of the recent ABI uses in this area, and for a minimal working start, we can essentially do the lock/unlocking at enter/exit of each CPython API we wrap.  A context manager used internally would look more aesthetic rather than pushing it onto callers, and we can talk about optimizing GIL lock/unlocking in the future (I have ideas here too).

Reply:
Yea, we started this documentation everything effort because I want to make changes like that. Please let us (the community) know what else we can do in #3661.